js/data/buffer.js

```diff
@@ -97,12 +93,27 @@ Buffer.prototype = {
         if (this.buffer) {
             gl.deleteBuffer(this.buffer);
         }
-    },
+    }
+};
+// End Buffer1 compatability code
 
-    setupViews: function() { /* intentional no-op */ }
 
+/**
+ * Set the attribute pointers in a WebGL context according to the buffer's attribute layout
+ * @private
+ * @param gl The WebGL context
+ * @param shader The active WebGL shader
+ * @param {number} offset The offset of the attrubute data in the currently bound GL buffer.
+ */
+Buffer.prototype.setAttribPointers = function(gl, shader, offset) {
+    for (var i = 0; i < this.attributes.length; i++) {
+        var attrib = this.attributes[i];
+
+        gl.vertexAttribPointer(
+            shader['a_' + attrib.name], attrib.components, gl[attrib.type.name],
+            false, this.itemSize, offset + attrib.offset);
```

Comment:
I'd say drop entirely. Simpler, less code, easier to understand. We can always reintroduce knew flexible methods later when we need them externally.